### DISCUSSION TITLE
Meshtastic messages not routed to/from MQTT

### DISCUSSION THREAD

**Discussion by amperpirat** [meshtastic/Discussions#82] [General]
Hi folks,
I just flashed my first nodes with meshtastics and they work as intended.
Now I wanted to integrate my fixed node A (connected to network using WiFi) into my home automation usining MQTT. Here's my setup:
Node B <-> Node A <-> MQTT

Messages between Node A and node B can be seen in any direction - no issues. Messages typed on node A are also routed to MQTT (MQTT explorer). But messages typed on node B are not tranferred to MQTT. 
Is there anything that I missed? Is this generally possible?

Here's my config on the devices:
- node A and node B share the same channel "mqtt" with the same PSK
- node A has MQTT configured to my private broker and has uplink/downlink enabled
- node A and node B are Heltec_v3 with firmware version 2.5.8.6485f03
- MQTT broker is mosquito 
- node A and node B are role CLIENT

If there's more info needed, please give me a shout.

Many thanks for your help and best regards,
Ole

**Comment by GUVWAF** [comment]
Node B has to have ["OK to MQTT"](https://meshtastic.org/docs/configuration/radio/lora/#ok-to-mqtt) setting enabled in the LoRa settings also.

**Comment by amperpirat** [comment]
thank you so much - that did the trick!

**Comment by venatiodecorus** [comment]
I'm trying to do almost the same thing here but I have been unable to get it to work. I'm using a Lilygo T-Echo and a Station G2, both with the most recent firmware.

My differences:
- I'm using the default public MQTT broker
- I'm using the default LongFast channel

My "base station" (Station G2) receives MQTT messages. Both nodes can communicate between themselves either way. And my T-Echo (no MQTT enabled, but Ok to MQTT is enabled) can send messages which get sent through the base station to the broker, so other MQTT nodes will reply.

HOWEVER, I do not receive any MQTT messages on the T-Echo. Only messages directly from my base station. Any suggestions? I've been trying to get this working for a minute now.

**Comment by garthvh** [comment]
The default channel is zero hopped so the messages will not be sent over Lora.
